common/reflect_resource.go

```diff
@@ -524,9 +538,17 @@ func DataToReflectValue(d *schema.ResourceData, r *schema.Resource, rv reflect.V
 
 func readReflectValueFromData(path []string, d attributeGetter,
 	rv reflect.Value, s map[string]*schema.Schema) error {
-	return iterFields(rv, path, s, func(fieldSchema *schema.Schema,
+	names := getJsonToFieldNameMap(rv.Type())
+	forceSendFields := []string{}
+	err := iterFields(rv, path, s, func(fieldSchema *schema.Schema,
 		path []string, valueField *reflect.Value) error {
 		fieldPath := strings.Join(path, ".")
+		alias := path[len(path)-1]
+		// GetOk returns false for default fields.
+		// Using GetOkExists determine which fields should be added to ForceSendFields.
+		if _, exists := d.GetOkExists(fieldPath); exists && isBasicType(fieldSchema.Type) {
```

Comment:
GetOkExists is technically deprecated, but there is no replacement and won't be removed until a replacement has been provided

https://github.com/hashicorp/terraform-plugin-sdk/pull/350#issuecomment-597888969